Library/Homebrew/caveats.rb

```diff
@@ -76,6 +76,7 @@ def keg_only_text
 
   def function_completion_caveats(shell)
     return unless keg
+    return if f.keg_only? # https://github.com/Homebrew/homebrew-core/pull/21941
```

Comment:
Would be good to make this print the correct paths instead.

Reply:
Would completion work for keg-only formulae though? The commands wouldn't (normally) be available.

Reply:
Above comment is correct. See the linked discussion in the core thread with ILZ. On the phone so will reply properly later from the laptop, but yeah, Alyssa is correct.

Reply:
Keg-only doesn't imply unlinked. We should explain that bash completion has been installed and where and, if needed, elaborate further on what's required to make it work.

Reply:
I'm happy to do it, but it will significantly bulk out this section of code given `zsh` and `fish` need covering too, all with slightly different sets of instructions. If you're happy with code bulk though, I'm happy to put the time in.

Reply:
I think it would be sufficient to just edit the paths noted below e.g. `#{HOMEBREW_PREFIX}/etc/bash_completion.d` in the keg-only case.

Reply:
Except that introduces a new failure model if people source it without handling the PATH issue at the same time, as outlined in https://github.com/Homebrew/homebrew-core/pull/21941#issuecomment-353217445.

Reply:
We are telling people where they are installed, though; I think that's preferable and if we have users who actually get confused by that and end up on our issue tracker: we can deal with that.

Reply:
Okay. Let's start fairly minimal and see if users report confusion/etc, that seems a reasonable approach, I agree.